src/clients/python/__init__.py

```diff
@@ -1218,31 +1234,45 @@ def _get_results(self, outputs, batch_size, request_id=None):
                         c_void_p(_crequest_get_shared_memory_handle_info(output_format[1], \
                                 byref(shm_addr), byref(shm_key), byref(shm_fd), \
                                 byref(offset), byref(byte_size))))
+                    if (np.prod(shape) * np.dtype(result_dtype).itemsize) < int(byte_size.value/batch_size):
```

Comment:
What is this? The size of the returned result must exactly match what is expected from the result shape.

Reply:
The output shared memory region may have a byte size of X but the size of the output not always be X, i.e. if the output shared memory region is too large we still want to allocate the appropriate number of byte size for each element in the batch to read.